File: java/android_efiles/app/src/main/java/efile/com/efiles/MainActivity.java
package efile.com.efiles;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import efile.com.efiles.fragments.EvidenceFragment;
import efile.com.efiles.fragments.ReportsFragment;
import efile.com.efiles.fragments.TransferFragment;

public class MainActivity extends AppCompatActivity {

    private TextView mTextMessage;
    private TabAdapter adapter;
    private TabLayout tabLayout;
    private ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewPager = findViewById(R.id.viewPager);
        tabLayout =  findViewById(R.id.tabLayout);
        adapter = new TabAdapter(getSupportFragmentManager());
        adapter.addFragment(new EvidenceFragment(),"Evidence");
        adapter.addFragment(new TransferFragment(), "Transfer");
        adapter.addFragment(new ReportsFragment(), "Reports");
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);
    }
}

File: kotlin/level_3_task_2/app/src/main/java/com/bliepy/studentportal/MainActivity.kt
package com.bliepy.studentportal

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*
import androidx.recyclerview.widget.GridLayoutManager



class MainActivity : AppCompatActivity() {

    private val data = arrayListOf<PortalData>()
    private val adapter = PortalAdapter(data)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        data.add(PortalData("VLO","https://vlo.informatica.hva.nl"))
        data.add(PortalData("SIS","https://sis.hva.nl"))
        data.add(PortalData("Roosters","https://rooster.hva.nl"))
        data.add(PortalData("HvA","https://hva.nl"))

        val recyclerView : RecyclerView = findViewById(R.id.portalList)
        recyclerView.setLayoutManager(GridLayoutManager(this, 2))
        recyclerView.setAdapter(adapter)

        fab.setOnClickListener { view ->
            val profileActivityIntent = Intent(this, AddPortalActivity::class.java)
        startActivity(profileActivityIntent)
    }
    }

}
